### PR TITLE
add anexia provider

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: add anexia provider

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note
add Anexia provider
```

### CODE REVIEW

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -987,8 +1001,17 @@ type NodeCloudSpec struct {
 	GCP          *GCPNodeSpec          `json:"gcp,omitempty"`
 	Kubevirt     *KubevirtNodeSpec     `json:"kubevirt,omitempty"`
 	Alibaba      *AlibabaNodeSpec      `json:"alibaba,omitempty"`
+	Anexia       *AnexiaNodeSpec       `json:"anexia,omitempty"`
 }
 
+// ProvisioningUtility specifies the type of provisioning utility.
+type ProvisioningUtility string
```
Comment (moadqassem) [inline pkg/api/v1/types.go]: Why introducing a new enum while you can use the one which was defined in the machine controller, in the flatcar package? 
```go
import "github.com/kubermatic/machine-controller/pkg/userdata/flatcar"

flatcar.ProvisioningUtility
```

File: pkg/machine/convert.go [hunk 2]
```diff
@@ -66,8 +67,15 @@ func GetAPIV1OperatingSystemSpec(machineSpec clusterv1alpha1.MachineSpec) (*apiv
 		if err := json.Unmarshal(decodedProviderSpec.OperatingSystemSpec.Raw, &config); err != nil {
 			return nil, fmt.Errorf("failed to parse flatcar config: %v", err)
 		}
+		// set ignition by default
+		provisioningUtility := apiv1.Ignition
```
Comment (moadqassem) [inline pkg/machine/convert.go]: Nit: The default provisioningUtility is ignition https://github.com/kubermatic/machine-controller/blob/master/pkg/userdata/flatcar/provider.go#L135-L145,  thus you don't need to specify that here.

File: pkg/resources/machine/common.go [hunk 3]
```diff
@@ -20,6 +20,8 @@ import (
 	"errors"
 	"fmt"
 
+	anexia "github.com/kubermatic/machine-controller/pkg/cloudprovider/provider/anexia/types"
```
Comment (moadqassem) [inline pkg/resources/machine/common.go]: Please move this import to be in the machine-controller import paths.

File: pkg/resources/machinecontroller/deployment.go [hunk 4]
```diff
@@ -239,6 +239,9 @@ func getEnvVars(data machinecontrollerData) ([]corev1.EnvVar, error) {
 		vars = append(vars, corev1.EnvVar{Name: "ALIBABA_ACCESS_KEY_ID", Value: credentials.Alibaba.AccessKeyID})
 		vars = append(vars, corev1.EnvVar{Name: "ALIBABA_ACCESS_KEY_SECRET", Value: credentials.Alibaba.AccessKeySecret})
 	}
+	if data.Cluster().Spec.Cloud.Anexia != nil {
+		vars = append(vars, corev1.EnvVar{Name: "ANXCLOUD_TOKEN", Value: credentials.Anexia.Token})
```
Comment (moadqassem) [inline pkg/resources/machinecontroller/deployment.go]: The env var should be `ANEXIA_TOKEN`

**Comment by zreigz** [comment]
@moadqassem thanks for the review, PTAL

**Pull request merged**